agent/agent.go

```diff
@@ -1188,6 +1198,9 @@ func (a *Agent) JoinLAN(addrs []string) (n int, err error) {
 	a.logger.Printf("[INFO] agent: (LAN) joining: %v", addrs)
 	n, err = a.delegate.JoinLAN(addrs)
 	a.logger.Printf("[INFO] agent: (LAN) joined: %d Err: %v", n, err)
+	if err == nil && a.joinLANNotifier != nil {
+		a.joinLANNotifier.Notify("READY=1")
```

Comment:
Do you want to log any non-`nil` error that's not `NotfyNoSocket` (since presumably that will be returned when not running under systemd? We could debug-level log for that one.